Workflow/Sources/Workflow/Models/AnyWorkflow.swift

```diff
@@ -0,0 +1,187 @@
+    }
+
+    @discardableResult public func launch(passedArgs: PassedArgs,
+                                          withLaunchStyle launchStyle: LaunchStyle = .default,
+                                          onFinish: ((Any?) -> Void)? = nil) -> LinkedList<AnyFlowRepresentable?>.Element? {
+        removeInstances()
+        instances = LinkedList(map { _ in nil })
+        var root: (instance: AnyFlowRepresentable, metadata: FlowRepresentableMetaData)?
+        var passedArgs = passedArgs
+
+        let metadata = first?.traverse { [self] nextNode in
+            let nextMetadata = nextNode.value
+            let flowRepresentable = nextMetadata.flowRepresentableFactory()
+            flowRepresentable.workflow = self
+            flowRepresentable.proceedInWorkflowStorage = { passedArgs = $0 }
+
+            let shouldLoad = flowRepresentable.shouldLoad(with: passedArgs)
+
+            defer {
+                guard let instance = instances.first?.traverse(nextNode.position) else { fatalError("Internal state of workflow completely mangled during launch.") }
+                let persistance = nextMetadata.calculatePersistance(passedArgs)
+                if shouldLoad {
+                    firstLoadedInstance = instance
+                    firstLoadedInstance?.value = flowRepresentable
+                    setupCallbacks(for: instance, onFinish: onFinish)
+                } else if !shouldLoad && persistance == .persistWhenSkipped {
+                    instance.value = flowRepresentable
+                    setupCallbacks(for: instance, onFinish: onFinish)
+                    orchestrationResponder?.launchOrProceed(to: (instance: instance, metadata: nextMetadata), from: convertInput(root))
+                    root = (instance: flowRepresentable, metadata: nextMetadata)
+                }
+            }
+            return shouldLoad
+        }?.value
+
+        guard let first = firstLoadedInstance,
+              let m = metadata else {
+            if let argsToPass = passedArgs.extract(nil) {
+                onFinish?(argsToPass)
+            }
+            return nil
+        }
+
+        orchestrationResponder?.launchOrProceed(to: (instance: first, metadata:m), from: convertInput(root))
+
+        return firstLoadedInstance
+    }
+
+    /// abandon: Called when the workflow should be terminated, and the app should return to the point before the workflow was launched
+    /// - Parameter animated: A boolean indicating whether abandoning the workflow should be animated
+    /// - Parameter onFinish: A callback after the workflow has been abandoned.
+    /// - Returns: Void
+    /// - Note: In order for this to function the workflow must have a presenter, presenters must call back to the workflow to inform when the abandon process has finished for the onFinish callback to be called.
+    public func abandon(animated: Bool = true, onFinish:(() -> Void)? = nil) {
```

Comment:
I'm not sure how to do this, but this abandon should not take in an animated parameter.